types/cls-hooked/index.d.ts

```diff
@@ -7,19 +7,21 @@
 
 import { EventEmitter } from 'events';
 
+export type Context = object;
+
 export interface Namespace {
-    active: any;
+    active: Context | null;
 
     set<T>(key: string, value: T): T;
     get(key: string): any;
-    run(fn: (...args: any[]) => void): void;
-    runAndReturn<T>(fn: (...args: any[]) => T): T;
-    runPromise<T>(fn: (...args: any[]) => Promise<T>): Promise<T>;
-    bind<F extends Function>(fn: F, context?: any): F; // tslint:disable-line: ban-types
+    run(fn: (context: Context) => void): Context;
+    runAndReturn<T>(fn: (context: Context) => T): T;
+    runPromise<T>(fn: (context: Context) => Promise<T>): Promise<T>;
```

Comment:
Should be `run(fn: (context?: Context) => void): Context`?

Reply:
I'm not very sure about it. Could you add the example code from https://github.com/Jeff-Lewis/cls-hooked/blob/master/README.md into test case?

Reply:
The documentation says _"The new context is passed as an argument to the callback when it's called."_ and the code backs it up.

We can declare `context` as optional or we can let it as it is now. The actual handler is not required to have it in its arguments list; it can be a function without arguments in both cases:

```typescript
const session = cls.createNamespace('my session');
const user = { id: 'foo' };
session.run(() => {
    session.set('user', user);
    session.run((context: cls.Context) => {
        console.log(`[With context] Find user using session.get(): ${session.get('user')}`);
        console.log(`[With context] Get user directly from the context received as argument: ${(<any>context).user}`);
    });
    session.run(() => {
        // The context is ignored by this handler; its content can still be accessed by using `session.get()`
        console.log(`[No context] Find user using session.get(): ${session.get('user')}`);
    });
});
```

I tested the code above. TypeScript does not complain.

Do you think it is better to mark it as optional in the declaration?

Reply:
If you have verified that tsc doesn't complain, then it's ok.